examples/examples.rs

```diff
@@ -82,37 +80,26 @@ fn examples() -> Result<()> {
     // This API handles variadics using Heterogeneous Lists.  This is one way to
     // call a function with multiple parameters:
 
-    print.call::<_, ()>(
-        hlist!["hello", "again", "from", "rust"],
-    )?;
+    print.call::<_, ()>(("hello", "again", "from", "rust"))?;
 
     // You can bind rust functions to lua as well
 
-    let check_equal = lua.create_function(|lua, args| {
-        // Functions wrapped in lua receive their arguments packed together as
-        // MultiValue.  The first thing that most wrapped functions will do
-        // is "unpack" this MultiValue into its parts.  Due to lifetime type
-        // signature limitations, this cannot be done automatically from the
-        // function signature, but this will be fixed with ATCs.  Notice the use
-        // of the hlist macros again.
-        let hlist_pat![list1, list2] = lua.unpack::<HList![Vec<String>, Vec<String>]>(args)?;
-
+    let check_equal = lua.create_function(|_, (list1, list2): (Vec<String>, Vec<String>)| {
```

Comment:
> So, you could make Lua itself take a lifetime parameter, and all closures would have to outlive the lifetime parameter? It would just be PhantomData in Lua? How do you get the lifetime into the callback, the callbacks can't know which Lua they're ultimately being called from except by getting the main state pointer. I might be misunderstanding what you're saying though.

I'm just taking a stab in the dark here. The playground links I posted demonstrate what I mean. The PhantomData isn't necessary, I used that to test things, `'a` is the lifetime outlived by all closures stored in the `Container`. Rust will pick the shortest one automatically (I think). I have no idea how this would look when applied to `Lua`.

> Which parameter needs to be kept, I'm not following.

The first argument (`&Lua`) passed to the closures, since there's no way to safely borrow it from the outside.